src/transformers/generation_logits_process.py

```diff
@@ -689,3 +689,72 @@ class LogitNormalization(LogitsProcessor, LogitsWarper):
     def __call__(self, input_ids: torch.Tensor, scores: torch.Tensor) -> torch.Tensor:
         scores = scores.log_softmax(dim=-1)
         return scores
+
+
+class SuppressBlank(LogitsProcessor):
+    r""" """
+
+    def __init__(self, tokenizer, sample_begin: int = 1):
+        self.tokenizer = tokenizer
+        self.sample_begin = sample_begin
+
+    def __call__(self, input_ids, scores):
+        tokens = input_ids
+        logits = scores
+        if tokens.shape[1] == self.sample_begin:
+            logits[:, self.tokenizer.encode(" ") + [self.tokenizer.eot]] = -np.inf
+        return logits
+
+
+class SuppressTokens(LogitsProcessor):
+    r""" """
+
+    def __init__(self, suppress_tokens):
+        self.suppress_tokens = list(suppress_tokens)
+
+    def __call__(self, input_ids, scores):
+        logits = scores
+        logits[:, self.suppress_tokens] = -np.inf
+        return logits
+
+
+class ApplyTimestampRules(LogitsProcessor):
+    r""" """
+
+    def __init__(self, tokenizer, sample_begin: int = 1, max_initial_timestamp_index: Optional[int] = None):
+        self.tokenizer = tokenizer
```

Comment:
We should also not pass the tokenizer here but instead all the attributes that are need further below (`no_timstamps`, `timestamp_begin`, ...)

Reply:
Totally !

Reply:
Actually just noticed that this is not needed for transcription necessarily - we can leave it out for now if you want!

Reply:
Let's actually delete this for now